cache/src/main/kotlin/com/dropbox/android/external/cache4/RealCache.kt

```diff
@@ -78,6 +81,8 @@ internal class RealCache<Key : Any, Value : Any>(
      */
     private val loadersSynchronizer = KeyedSynchronizer<Key>()
 
+    private var totalWeight = 0;
```

Comment:
```suggestion
    private val totalWeight = atomic(0)
```
Unlike guava we don't use a coarse grained lock (e.g. in `put()` and `get()`) to guard concurrent read write.

I'm also thinking if it's worth following caffeine's approach where expiry and evictions are treated as "maintenance" activities which are performed asynchronously, so concurrent ops can be much faster and we don't need to care about syncing the main `ConcurrentHashMap` and the other access order queues too much?